libs/platform/src/lib/form/form-group/services/form-field-layout.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject } from 'rxjs';
+import { ColumnLayout } from '@fundamental-ngx/platform/shared';
+import { distinctUntilChanged, startWith } from 'rxjs/operators';
+import { normalizeColumnLayout } from '../helpers';
+
+type Layouts = {
+    labelColumnLayout: ColumnLayout;
+    fieldColumnLayout: ColumnLayout;
+    gapColumnLayout: ColumnLayout;
+};
+
+@Injectable()
+export class FormFieldLayoutService {
+    readonly needsInlineHelpPlace: Observable<boolean>;
+    private _needsInlineHelpPlace$ = new ReplaySubject<boolean>(1);
+    private _elementsMap: Map<Record<string, any>, boolean> = new Map();
+    private _needsInlineHelpPlace = false;
+
+    constructor() {
+        this.needsInlineHelpPlace = this._needsInlineHelpPlace$.pipe(startWith(false), distinctUntilChanged());
+    }
+
+    setNeedsInlineHelp(target: Record<string, any>, isNeeded: boolean): void {
+        this._elementsMap.set(target, isNeeded);
+        this._updateCombinedValue();
+    }
+
+    getFixedLayouts(layouts: Layouts): Layouts {
+        const { fieldColumnLayout, labelColumnLayout, gapColumnLayout } = {
```

Comment:
do you really need to create an object and then spread it?

Reply:
Probably no, initially had different idea. Thanks